docs/naps/8-telemetry.md

```diff
@@ -170,6 +170,334 @@ In napari itself, the following changes should be implemented:
+
+    def __init__(self, file_path):
+        if file_path in self._instances:
+            raise ValueError(f"Already exists {file_path}")
+        self._instances[file_path] = self
+        self.file_path = file_path
+        self.data = {"week_info": get_week_identifier()}
+        if os.path.exists(self.file_path):
+            with open(self.file_path, "r") as fp:
+                try:
+                    data_ = json.load(fp)
+                except ValueError:
+                    pass
+                else:
+                    if data_.get("week_info") == self.data["week_info"]:
+                        self.data = data_
+
+    def save(self):
+        with open(self.file_path, "w") as fp:
+            json.dump(self.data, fp)
+
+    @classmethod
+    def get_collector(cls, file_path):
+        file_path = str(file_path)
+        if file_path not in cls._instances:
+            cls(file_path)
+        return cls._instances[file_path]
+
+    @staticmethod
+    def extract_size_info_volumetric(data):
+        return {
+            "size": data.size,
+            "shape": data.shape,
+            "dtype": str(data.dtype),
+            "ndim": data.ndim,
+            "size_category": SizeCategory.from_size(data.size).name,
+            "class": data.__class__.__name__,
+        }
+
+    @staticmethod
+    def extract_size_info_spatial(data):
+        return {
+            "size": data.size,
+            "shape": data.shape,
+            "dtype": str(data.dtype),
+            "ndim": data.shape[1],
+            "size_category": SizeCategory.from_size(data.shape[0]).name,
+            "class": data.__class__.__name__,
+        }
+
+    def add_data_info(self, array, layer_name_str):
+        if layer_name_str in {"labels", "image"}:
+            size_info = self.extract_size_info_volumetric(array)
+        else: # layer_name_str in {"points", "shapes", "vectors"}:
+            size_info = self.extract_size_info_spatial(array)
+        key = str((layer_name_str, size_info["size_category"], size_info["ndim"], size_info["class"], size_info["dtype"]))
+        if key in self.data:
+            self.data[key]["count"] += 1
+        else:
+            self.data[key] = {"count": 1}
+
+        self.save()
+
+
+def get_full_information():
+    middle_info = get_middle_information()
+    data_collector = DataSizeTypeCollector.get_collector(
+        Path(appdirs.user_config_dir("napari")) / "data_size_info.json"
+    )
+    # app_model = get_app() - there is a need to add a collection of data to app_model
+    return {
+        **middle_info,
+        "data_size_info": data_collector.data,
+        "command_usage": { # it should be collected from app_model
+            'napari:window:file:open_files_dialog': 50,
+            'napari:window:file:copy_canvas_screenshot': 10,
+            'napari:window:file:restart': 1,
+            'napari:window:file:close': 100,
+        },
+    }
+```
+
+Code for generate sample data (requires above code):
```

Comment:
Again maybe expand a bit:

```suggestion
The script below generates sample reports for each level of information gathered. The information about layer data is mocked so that this script can be executed independently of `napari`.
```